### ISSUE TITLE
com.github.dockerjava.core.command.PullImageResultCallback is deprecated and replaced by which class?

### ISSUE THREAD

**Issue by WallaceSUI** [docker-java/docker-java#2179]
Hey developers,

Sorry to trouble you. Recently I wanna customize my own `PullImageResultCallback` class to add and expose more levels of logs for image pulling. But I have found that `com.github.dockerjava.core.command.PullImageResultCallback` is marked as @deprecated. I searched for java-doc and github, there are not many comments explanation for this. I'm not sure it is replaced by which method?

1: From my searching, I found that `com.github.dockerjava.api.command.PullImageResultCallback` has nearly the same codes but under API package. I wonder whether this is the new class for the same functionality?

2: Also, I wonder if I wanna customize this class to expose more logs for image pulling, whether I can only get more strings/info from the class `ResponseItem` below?  There are no more info I can get for image pulling except this class `ResponseItem`? Thanks.
```java
public class ResponseItem implements Serializable {
    private static final long serialVersionUID = -5187169652557467828L;
    @JsonProperty("stream")
    private String stream;
    @JsonProperty("status")
    private String status;
    @JsonProperty("progressDetail")
    private ResponseItem.ProgressDetail progressDetail;
    /** @deprecated */
    @Deprecated
    @JsonProperty("progress")
    private String progress;
    @JsonProperty("id")
    private String id;
    @JsonProperty("from")
    private String from;
    @JsonProperty("time")
    private Long time;
    @JsonProperty("errorDetail")
    private ResponseItem.ErrorDetail errorDetail;
    /** @deprecated */
    @Deprecated
    @JsonProperty("error")
    private String error;
    @JsonProperty("aux")
    private ResponseItem.AuxDetail aux;

``` 

Thanks a lot.

**Comment by WallaceSUI** [comment]
@bsideup @kiview 
Thanks.

**Comment by Derppening** [comment]
> 1: From my searching, I found that com.github.dockerjava.api.command.PullImageResultCallback has nearly the same codes but under API package. I wonder whether this is the new class for the same functionality?

Yes, at least that is what I use in my own projects as well. It seems to stem from commit 9be9c5e.

> 2: Also, I wonder if I wanna customize this class to expose more logs for image pulling, whether I can only get more strings/info from the class ResponseItem below?

I am not sure what information you're referring to, but my understanding is that `PullResponseItem` represents a single message from Docker as exposed by docker-java.
